README.md

```diff
@@ -1,121 +1,229 @@
 Make sure to record the environmental variables (lines that contains the "export" commands). They are needed for running PDC and to use the libraries again.
-# Install libfabric
+
+## Preparing for Installation
+
+PDC relies on [`libfabric`](https://github.com/ofiwg/libfabric/) as well as [`mercury`](https://github.com/mercury-hpc/mercury). Therefore, let's **prepare the dependencies**.
+### Preparing Work Space
+
+Before installing the dependencies and downloading the code repository, we assume there is a directory created for your installation already, e.g. `$WORK_SPACE` and now you are in `$WORK_SPACE`.
+
+```bash
+export WORK_SPACE=/path/to/your/work/space
+mkdir -p $WORK_SPACE/source
+mkdir -p $WORK_SPACE/install
+```
+
+### Download Necessary Source Repository
+
+Now, let's download [`libfabric`](https://github.com/ofiwg/libfabric/), [`mercury`](https://github.com/mercury-hpc/mercury) and [`pdc`](https://github.com/hpc-io/pdc/tree/develop) into our `source` directory.
+
+```bash
+cd $WORK_SPACE/source
+git clone git@github.com:ofiwg/libfabric.git
+git clone git@github.com:mercury-hpc/mercury.git
+git clone git@github.com:hpc-io/pdc.git
+```
+
+### Prepare Directories for Artifact Installation
+```bash
+export LIBFABRIC_SRC_DIR=$WORK_SPACE/source/libfabric
+export MERCURY_SRC_DIR=$WORK_SPACE/source/mercury
+export PDC_SRC_DIR=$WORK_SPACE/source/pdc
+
+export LIBFABRIC_DIR=$WORK_SPACE/install/libfabric
+export MERCURY_DIR=$WORK_SPACE/install/mercury
+export PDC_DIR=$WORK_SPACE/install/pdc
+
+mkdir -p $LIBFABRIC_SRC_DIR
+mkdir -p $MERCURY_SRC_DIR
+mkdir -p $PDC_SRC_DIR
+
+mkdir -p $LIBFABRIC_DIR
+mkdir -p $MERCURY_DIR
+mkdir -p $PDC_DIR
+
+echo "export LIBFABRIC_SRC_DIR=$LIBFABRIC_SRC_DIR" > $WORK_SPACE/pdc_env.sh
+echo "export MERCURY_SRC_DIR=$MERCURY_SRC_DIR" >> $WORK_SPACE/pdc_env.sh
+echo "export PDC_SRC_DIR=$PDC_SRC_DIR" >> $WORK_SPACE/pdc_env.sh
+
+echo "export LIBFABRIC_DIR=$LIBFABRIC_DIR" >> $WORK_SPACE/pdc_env.sh
+echo "export MERCURY_DIR=$MERCURY_DIR" >> $WORK_SPACE/pdc_env.sh
+echo "export PDC_DIR=$PDC_DIR" >> $WORK_SPACE/pdc_env.sh
 ```
-wget https://github.com/ofiwg/libfabric/archive/v1.11.2.tar.gz
-tar xvzf v1.11.2.tar.gz
-cd libfabric-1.11.2
-mkdir install
-export LIBFABRIC_DIR=$(pwd)/install
 
+Remember, from now on, at any time, you can simply run the following to set the above environment variables so that you can run any of the following command for your installation.
+
+```bash
+export WORK_SPACE=/path/to/your/work/space
+source $WORK_SPACE/pdc_env.sh
+```
+
+### Compile and Install`libfabric`
+
+Check out tag `v1.11.2` for `libfabric`:
+
+```bash
+cd $LIBFABRIC_SRC_DIR
+git checkout tags/v1.11.2
+```
+
+Configure, compile and install:
+
+```bash
 ./autogen.sh
-./configure --prefix=$LIBFABRIC_DIR CC=gcc CFLAG="-O2"
-make -j8
+./configure --prefix=$LIBFABRIC_DIR --disable-efa --disable-sockets CC=cc CFLAG="-O2"
```

Comment:
Can you add a note here that "--disable-efa --disable-sockets" are only needed on Cori/Perlmutter